src/transformers/models/timm_wrapper/modeling_timm_wrapper.py

```diff
@@ -0,0 +1,215 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+from torch import Tensor, nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from ...modeling_outputs import BaseModelOutput, ImageClassifierOutput
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    add_start_docstrings_to_model_forward,
+    is_timm_available,
+    requires_backends,
+)
+from .configuration_timm_wrapper import TimmWrapperConfig
+
+
+if is_timm_available():
+    import timm
+
+
+TIMM_WRAPPER_INPUTS_DOCSTRING = r"""
+    Args:
+        pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Pixel values. Pixel values can be obtained using [`AutoImageProcessor`]. See [`ViTImageProcessor.__call__`]
+            for details.
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. Not compatible with timm wrapped models.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. Not compatible with timm wrapped models.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        **kwargs:
+            Additional keyword arguments passed along to the model forward.
+"""
+
+
+def _load_timm_model(config: TimmWrapperConfig, add_classification_head: bool = False):
+    # timm model will not add classification head if num_classes = 0
+    num_classes = config.num_labels if add_classification_head else 0
+    model = timm.create_model(model_name=config.architecture, pretrained=False, num_classes=num_classes)
+    return model
+
+
+class TimmWrapperPreTrainedModel(PreTrainedModel):
+    main_input_name = "pixel_values"
+    supports_gradient_checkpointing = False
+    config_class = TimmWrapperConfig
+    base_model_prefix = "model"
+    _no_split_modules = []
+
+    def __init__(self, *args, **kwargs):
+        requires_backends(self, ["vision", "timm"])
+        super().__init__(*args, **kwargs)
+
+    @staticmethod
+    def _fix_state_dict_key(key):
+        """
+        Override original method which renames `gamma` and `beta` to `weight` and `bias`.
+        We don't want this behavior for timm wrapped models. Instead, this method adds
+        "timm_model." prefix to be able to load official timm Hub checkpoints.
+        """
+        if "timm_model." not in key:
+            return f"timm_model.{key}"
+        return key
+
+    def _init_weights(self, module):
+        """
+        Empty init weights function to ensure compatibility of the class in the library.
+        """
+        if isinstance(module, (nn.Linear)):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
```

Comment:
it's not clear why this is here, what it's attempting to do. timm has model specific init fns though they aren't separately callable right now, doing something like this that could overwrite timm defaults would change model behaviour

Reply:
Added this to initialize classifier, without this weights are not properly initialized, probably due to how the model is created in transformers

```python
from transformers import TimmWrapperForImageClassification

# --------------
# With init
# --------------

model = TimmWrapperForImageClassification.from_pretrained("timm/resnet18.a1_in1k", num_labels=10, ignore_mismatched_sizes=True)
print(model.timm_model.fc.weight[:3, :3])
# tensor([[-0.2117, -0.2422, -0.2540],
#         [-0.1106, -0.1856, -0.0152],
#         [-0.3430, -0.6446, -0.0530]], grad_fn=<SliceBackward0>)


# --------------
# Without init
# --------------

# patch with empty init weight function to check
def empty_init(self, module):
    pass
TimmWrapperForImageClassification._init_weights = empty_init

model = TimmWrapperForImageClassification.from_pretrained("timm/resnet18.a1_in1k", num_labels=10, ignore_mismatched_sizes=True)
print(model.timm_model.fc.weight[:3, :3])
# tensor([[0., 0., 0.],
#         [0., 0., 0.],
#         [0., 0., 0.]], grad_fn=<SliceBackward0>)
```

Ideally, we should get rid of this, but it's not common for transformers to load external models, so it might require more code changes. For now, its a simple fix to enable model loading with initialized classifier if shapes are mismatched.